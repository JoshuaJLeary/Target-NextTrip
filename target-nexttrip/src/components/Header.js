import React, { Component } from 'react';

class Header extends Component {
  render() {
    return (
      //this header contains the Metro Transit image and image banner
      <div className='headerContainer'>
          <div id='metroImgContainer'>
            <img src='https://www.metrotransit.org/images/mob_logo.png' className='metroImg' alt='metroTransitLogo' />
          </div>
          <div id='metroImgBannerContainer'>
            <img src='https://www.metrotransit.org/images/mob_banner.png' className='metroImgBanner' alt='metroTransitBanner' />
          </div>
        </div>
    )
  }
}

export default Header;
