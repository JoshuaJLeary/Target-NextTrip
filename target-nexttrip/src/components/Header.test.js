import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header></Header>, div);
});

it('renders Header correctly', () => {
 const { getByTestId } = render(<Header></Header>);
 expect(getByTestId('header')).toHaveTextContent('');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Header></Header>).toJSON();
  expect(tree).toMatchSnapshot();
})