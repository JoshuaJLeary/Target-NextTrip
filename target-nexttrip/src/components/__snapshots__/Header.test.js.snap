// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="headerContainer"
  data-testid="header"
>
  <div
    id="metroImgContainer"
  >
    <img
      alt="metroTransitLogo"
      className="metroImg"
      src="https://www.metrotransit.org/images/mob_logo.png"
    />
  </div>
  <div
    id="metroImgBannerContainer"
  >
    <img
      alt="metroTransitBanner"
      className="metroImgBanner"
      src="https://www.metrotransit.org/images/mob_banner.png"
    />
  </div>
</div>
`;
